packages/algob/src/internal/util/lists.ts

```diff
@@ -0,0 +1,27 @@
+
+function getLast<T> (ts: T[]): T {
+  return ts[ts.length - 1];
+}
+
+function getLastSubchild<T> (ts: T[][]): T {
```

Comment:
Please add a comment or rename.
It's not clear without reading the code what the function is doing (what is a subchild)?

Reply:
To be honest I don't have good proposal here ;)

Reply:
```

export function partitionByFn<T> (
  f: (s: T, s1: T) => boolean,
  input: T[]
): T[][] {
  if (input.length === 0) {
    return [];
  }
  var out = [[input[0]]];
  let last = input[0];
  for (const current of input.slice(1)) {
    if (f(last, current)) {
      out.push([current]);
    } else {
      out[out.length - 1].push(current);
    }
    last = current;
  }
  return out;
}
```

Reply:
can we change it to something like this ?
and remove the getLast and getLastSubChild functions ?

Reply:
will this work ?

Reply:
It will work but it will be even harder to understand what it does.
And it will be faster.